### DISCUSSION TITLE
aggrid conditional formatting

### DISCUSSION THREAD

**Discussion by tonydero** [zauberzeug/nicegui#754] [Q&A] [answered]
### Question

Hello!

Is there any way to apply conditional formatting to an `aggrid` object in nicegui? I would like to change the text color based on the value.

It seems it's possible [in js](https://blog.ag-grid.com/conditional-formatting-for-cells-in-ag-grid/), but I have no idea how I would go about translating that to nicegui. I'm not very familiar with js to begin with either unfortunately (well, frontend design in general). Would I need to use `call_api_method` to accomplish this?

Any tips/pointers would be most welcome, thanks!

**Comment by falkoschindler** [comment, marked as answer]
Good question, @tonydero!

Setting grid options including callable JavaScript functions like `'cellClassRules': {'cell-fail': 'x => x.value < 21', 'cell-pass': 'x => x.value >= 21'}` is currently not possible. But I found a snippet of JavaScript code to set those rules as soon as the client connects:

```py
from nicegui import app, ui

ui.add_head_html('''
<style>
.cell-fail { background-color: #f44336; }
.cell-pass { background-color: #4caf50; }
</style>
''')

grid = ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
})

async def format() -> None:
    await ui.run_javascript(f'''
        getElement({grid.id}).gridOptions.columnApi.getColumn("age").getColDef().cellClassRules = {{
            "cell-fail": x => x.value < 21,
            "cell-pass": x => x.value >= 21,
        }};
        getElement({grid.id}).gridOptions.api.refreshCells();
    ''', respond=False)

app.on_connect(format)

ui.run()
```

**Reply by tonydero** [reply]
Thanks! I can work with this.
